### PR TITLE
switch mpileup to bcftools and subset to non-ref

### PR DESCRIPTION
**Pull request by nickhsmith**
the mpileup mechanism was outdated and no longer matched the samtools/bcftools practices for variant calling. I updated the variant calling modules using the nf-core/modules `bcftools/mpileup` module.

I am waiting for the nf-core/modules/bcftools to be updated (PR pending) to update officially.

### CODE REVIEW

File: conf/modules.config [hunk 1]
```diff
@@ -1135,31 +1146,31 @@ process{
...
     // CONTROLFREEC
     if (params.tools && (params.tools.split(',').contains('controlfreec') || params.tools.split(',').contains('mpileup'))) {
-        withName: 'NFCORE_SAREK:SAREK:BAM_VARIANT_CALLING_GERMLINE_ALL:BAM_VARIANT_CALLING_MPILEUP:SAMTOOLS_MPILEUP' {
+        withName: 'NFCORE_SAREK:SAREK:BAM_VARIANT_CALLING_GERMLINE_ALL:BAM_VARIANT_CALLING_MPILEUP:BCFTOOLS_MPILEUP' {
             ext.prefix = { meta.num_intervals <= 1 ? "${meta.id}.normal" : "${meta.id}_${intervals.simpleName}.normal" }
         }
 
         withName: 'NFCORE_SAREK:SAREK:BAM_VARIANT_CALLING_GERMLINE_ALL:BAM_VARIANT_CALLING_MPILEUP:CAT_MPILEUP' {
-            ext.prefix = { "${meta.id}.normal.mpileup.gz" }
+            ext.prefix = { "${meta.id}.all.mpileup.gz" }
```
[the remark this answers is not in the document]
Comment (maxulysse) [reply conf/modules.config:1154]: Yes, but in this one, it's the germline variant calling that is done. So let's keep it normal

**Comment by maxulysse** [comment]
Should SAMTOOLS_PILEUP be removed as well?
I'm assuming CAT_CAT should be too?

**Comment by maxulysse** [comment]
Will it impact in anyway the controlfreec part?

**Comment by FriederikeHanssen** [comment]
Is this one superseded by #832 ?

**Pull request closed without merging**